gateway/src/main/java/io/zeebe/gateway/impl/broker/BrokerClientImpl.java

```diff
@@ -97,8 +103,35 @@ public BrokerClientImpl(final GatewayCfg configuration, final ActorClock actorCl
     transport = transportBuilder.build();
     internalTransport = internalTransportBuilder.build();
 
+    final String[] parts = configuration.getCluster().getContactPoint().split(":");
+    final Integer port = Integer.parseInt(parts[1]);
+
+    atomix =
+        AtomixCluster.builder()
+            .withMemberId("gateway")
+            .withClusterId(configuration.getCluster().getClusterName())
+            .withMembershipProvider(
+                BootstrapDiscoveryProvider.builder()
+                    .withNodes(Address.from(parts[0], (port + CLIENT_ADDR_OFFSET)))
+                    .build())
+            .build();
+    Log.info("Gateway address {}", configuration.getNetwork().toString());
+
     topologyManager =
         new BrokerTopologyManagerImpl(internalTransport.getOutput(), this::registerEndpoint);
+    atomix.getMembershipService().addListener(topologyManager);
```

Comment:
There could be a race condition between completion of the join future and the invocation of the membership listener. Maybe it would be more safe to only register the listeners after we joined the cluster? Not sure about this one.

Reply:
Yeah that was the part I wasn't sure about, I'm fine with keeping it for now this way.